libs/storage/Tsavorite/cs/src/core/Index/Recovery/Recovery.cs

```diff
@@ -1126,7 +1170,8 @@ public async ValueTask RestoreHybridLogAsync(long beginAddress, long headAddress
                     tailPage = GetPage(fromAddress);
                     headPage = GetPage(headAddress);
 
-                    recoveryStatus = new RecoveryStatus(GetCapacityNumPages(), tailPage, untilAddress, 0);
+                    var capacity = GetCapacityNumPages();
+                    recoveryStatus = new RecoveryStatus(capacity, capacity - MinEmptyPageCount, tailPage, untilAddress, 0);
```

Comment:
It'd be a little cleaner to pass MinEmptyPageCount as the second param and do the subtraction in the ctor